### ISSUE TITLE
Add labels to a ticket

### ISSUE THREAD

**Issue by kylynnloi** [aehrc/snomio#166]
**As a:** Terminology Author,  
**I want to:**  add labels to a ticket
**So I can:**  use the ticket system to manage workflow

---

Users can use labels to manage priority and/or group tickets together

### Acceptance Criteria

1. A ticket can have more than one label, a label can be reused for multiple tickets
2. User can create new labels or use precreated ones
3. User can add a label to a ticket from the table list or from within the detailed view of the ticket
4. Labels should be searchable/filterable in the table list of tickets.

**Comment by clinnygee** [comment]
Hey. I would like to move 2. and 4. To a different ticket or two different tickets. 

I'm also not sure how I should implement 2. and where. 

?

4. is a lot of work and deffs needs it's own ticket

**Comment by kylynnloi** [comment]
@clinnygee we can currently seach labels already right? using the search bar - it's the filters that don't work?

**Comment by clinnygee** [comment]
@kylynnloi Yes exactly, like labels.name:whatever. Un an kind of related note next to the search thing I think I should add a button - just because that one in particular is a slow query and can make a bit of funky behavior (not unexpected behavior, just laggy) happen. So it doesn't update when you type, just when you hit search.

**Comment by kylynnloi** [comment]
@clinnygee happy to move 2 to another ticket.
i think 4 is a general ticket to fix filter so it works/useful and raise a separate ticket to have a search button to start the search?

**Comment by kylynnloi** [comment]
@clinnygee happy to take number 4 out, but can we please remove the filter functionality from the table (until we fix it) to avoid confusion

@ChrisKellaleaMaynard and I will update the ticket

**Comment by HymaKanchiAU** [comment]
@clinnygee , Unable to see whether the filter is re-added 

1. Clicked on a ticket https://dev-snomio.ihtsdotools.org/dashboard/tickets/individual/2 which had the following labels to start with JiraExport

The_Peter_Doherty_Institute_for_Infection_and_Immunity
Fred_IT
Pfizer
Guidance

2. Clicked on Edit and unticked The_Peter_Doherty_Institute_for_Infection_and_Immunity
Fred_IT
Pfizer
3. Tried to click on the checkbox The_Peter_Doherty_Institute_for_Infection_and_Immunity

I am unable to see "The_Peter_Doherty_Institute_for_Infection_and_Immunity" in the Labels section 
Once I refresh the page I can see that the label has been added.  Also, noticed that due to this any new labels added after that are also not showing but after a refresh they are available.

**Comment by HymaKanchiAU** [comment]
Assign multiple labels to a ticket from Backlog Page = Pass. Note have to refresh the Edit ticket page to view the changes if the page is already loaded.

Assign multiple labels to a ticket from Edit Ticket Page =Pass

Assign label to a ticket from Backlog Page = Pass

Unassign label from a ticket in Backlog Page = Pass

Unassign label from Edit ticket page = Pass Note have to refresh View ticket page for the unassign label to disappear.


Compare the labels from JIRA is pending as the filer is still work in progress.
- Used the query to find if there are any tickets other than the labels that were imported = Pass
- project = "AMT Authoring" AND "External Requester" not in ("NSW_Health","Arrotex","Pfizer","External","SA_Health","The_Peter_Doherty_Institute_for_Infection_and_Immunity","Fred_IT",
"HealthDirect","Kind_Medical","Omnicell","AlfredHealth","ORACLE","Eli_Lilly","ArumaLabs","AFTPharmaceuticals","PBS","EchoTherapeutics","MedicalDirector","Internal","GenericHealth","Sun_Pharma","KMCHealthCare","AstraZeneca","Synevi","BlackTriangle","Chemist_Warehouse","HealthShare_Victoria","Cerner","EgoPharmaceuticals","DoH","Accord","FreseniusKabi","DrReddys","JunoPharmaceuticals","Viatris","Petrus","MedRedleaf","Tilray","AllergyConceptsPtyLtd","ArtemHealth","Janssen","Aspen","GedeonRichterAustralia","MIMs","WoltersKluwer","Pharmacor","InterPharma","Sanofi","AMT-NPC","Arrotex","Thema","ChiesiAustralia","MolnlyckeHealthCare","CR_HTA_Consulting","Sandoz","IlluminateHealthConsulting","EzyMed","Takeda","Mylan","Boehringer-Ingelheim","PharmacyHealthcare","E","Halmed","KeyPharmaceuticals","PharmHos","BesinsHealthcare","Kluwer","Wolters","GSK","AlfredHealthJun19","LinkHealthcare","BBraunMay19","SAHealthJun19","PetrusMar19","HalmedMar19","GSKMay19","FresensiusKabiApr19","DVAFeb2019","SAHealthDec18","GenericHealthApr19","EzymedMay19","Phebra","NovaPharmaApr19","AspenJun19","DispenseWorksApr19","ArrowPharma","ERRCD_QLD_gap_list","CarePharmApr19","SandozMay19","BrandwoodCKC","ViforJun19","EgoApr19","PharmacorMay19","SunPharmaApr19","Bayer","SanofiMay19","POSWorksJun19","GenericHealthMay19","Apotex","PBSMay19","CelgeneAusApr19","CernerApr19","OmnicellJun19","HealthcareSoftware","PharmAustApr19","NPC","Guidance","FREDITOct2018","HealthDirectOct2018") and displayed 0 results.

![image](https://github.com/aehrc/snomio/assets/126848487/ce430f1a-f480-4456-a091-b54afc81c425)
